app/presenters/notices/setup/abstraction-alerts/alert-thresholds.presenter.js

```diff
@@ -19,12 +19,31 @@ function go(session) {
     backLink: `/system/notices/setup/${session.id}/abstraction-alerts/alert-type`,
     caption: session.monitoringStationName,
     pageTitle: 'Which thresholds do you need to send an alert for?',
-    thresholdOptions: _thresholdOptions(session.licenceMonitoringStations, session.alertThresholds)
+    thresholdOptions: _thresholdOptions(session.licenceMonitoringStations, session.alertType, session.alertThresholds)
   }
 }
 
-function _thresholdOptions(licenceMonitoringStations, alertThresholds = []) {
-  return licenceMonitoringStations.map((licenceMonitoringStation) => {
+/**
+ * Filters licence monitoring stations based on alert type.
+ *
+ * When the alert type is not 'stop' or 'reduce', all stations with any restriction are included.
+ *
+ * @private
+ */
+function _relevantLicenceMonitoringStations(stations, alertType) {
+  return stations.filter((station) => {
+    if (alertType === 'stop' || alertType === 'reduce') {
+      return station.restriction_type === alertType
+    }
+
+    return Boolean(station.restriction_type)
+  })
+}
```

Comment:
Could I suggest this as an alternative (assuming the tests still pass)?

```suggestion
function _relevantLicenceMonitoringStations(licenceMonitoringStations, alertType) {
  if (alertType !== 'stop' && alertType !== 'reduce') {
    return licenceMonitoringStations
  }

  return licenceMonitoringStations.filter((licenceMonitoringStation) => {
    return licenceMonitoringStation.restriction_type === alertType
  })
}
```

In this version we only iterate the `licenceMonitoringStations` if the alert type is 'stop' or 'reduce' (rather than always). Plus we avoid the use of `Boolean()` which [MSDN says](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean/Boolean)

> Warning: You should rarely find yourself using `Boolean` as a constructor.